### DISCUSSION TITLE
Accessing files over http

### DISCUSSION THREAD

**Discussion by NiKiZe** [winfsp/winfsp#383] [General]
I would like to access files over http, this might be a premature question since I have yet to do proper research...

I have seen [some references to httpfs](https://softwarerecs.stackexchange.com/questions/30324/mount-iso-over-http), but not yet found anything in regards to this on windows.

I would like **one of** these:
* Mount a ISO over http, range requests being done as needed - (similar to how @iPXE does access with `sanhook`/`sanboot` over http)
* Mount some other fs from an image over http
* Mount some http index, guess that is closest to how httpfs does it - but again not done my research here yet.

I want this to work in WinPE - the use-case is to access `sources` directory of ISO or windows installation media, without having anything other than internet access (and blocked RPC ports)

If anyone have suggestions or ideas while I continue research that would be greatly apricated.

**Comment by billziss-gh** [comment]
WinFsp is low-level software that allows developers to create file systems. WinFsp is not a file system itself. (Think of it like this: WinFsp is a file system "maker", and not a file system.)

A file system could conceivably provide the functionality you are looking for and such a file system could be written on top of WinFsp. While there are many file systems that use HTTP, I am not aware of one that provides the exact functionality that you want.

**Reply by NiKiZe** [reply]
Thank you, got most of that.

Probably with https://github.com/billziss-gh/winspd or similar would be the better approach since we are talking about a "disk image" (?) - Note to self: test if it does ISO mount.

**Reply by billziss-gh** [reply]
WinSpd is also low-level software that allows developers to create "storage devices" (i.e. devices that look like "disk drives" to the operating system). So yes, this may be closer to what you need.

**Reply by NiKiZe** [reply]
Sorry for the delay in getting back to this.. can WinSpd in its current form expose as optical device as well? For my usecase a disk device would probably expose fine, but just in case I would like to know how close it could get. (Just in case WinPE, or windows setup would start behaving badly for some reason)

**Reply by billziss-gh** [reply]
I am not sure what you mean by "optical device". If you are asking if it can present a device that looks like a CDROM/DVDROM/etc. I do not believe so (at least not without additional work).

**Reply by NiKiZe** [reply]
Yes optical device = cd/dvd, thanks for always taking the time to answer everyone's questions, very appreciated!
